shub/config.py

```diff
@@ -370,9 +371,34 @@ def get_image(self, target):
         return image
 
 
-Target = namedtuple('Target', ['project_id', 'endpoint', 'apikey', 'stack',
-                               'image', 'requirements_file', 'version',
-                               'eggs'])
+_Target = namedtuple('Target', ['project_id', 'endpoint', 'apikey', 'stack',
+                                'image', 'requirements_file', 'version',
+                                'eggs'])
+
+
+class Apikey(str):
+
+    def __new__(cls, *args, **kwargs):
+        cls._inst = super(Apikey, cls).__new__(cls, *args, **kwargs)
+        return cls._inst
+
+    def __init__(self, apikey=None):
+        self.apikey = apikey
+
+    def __repr__(self):
+        if self.apikey:
+            N = len(self.apikey)
+            k = N // 5 + 1
```

Comment:
I set N to be variable to avoid displaying all the characters if the string was short. But if is fixed to 32-chars, it makes more sense to use a constant value.